src/main/java/com/synopsys/integration/jenkins/JenkinsProxyHelper.java

```diff
@@ -74,8 +73,8 @@ public JenkinsProxyHelper() {
         this.isBlank = true;
     }
 
-    public static JenkinsProxyHelper fromJenkins(Jenkins jenkins) {
-        ProxyConfiguration proxyConfig = Optional.ofNullable(jenkins)
+    public static JenkinsProxyHelper fromJenkins(JenkinsWrapper jenkinsWrapper) {
+        ProxyConfiguration proxyConfig = Optional.ofNullable(jenkinsWrapper.getJenkins()).orElse(null)
```

Comment:
Given the hoops we have to jump through to test this aspect of the class, it may be worthwhile to move this into the JenkinsWrapper and just have a method on JenkinsWrapper to get the ProxyConfiguration out of its internal Jenkins.

Reply:
So I agree that moving the logic which handles retrieving the ProxyConfiguration from Jenkins should be in JenkinsWrapper.getProxyConfiguration(). But, I think the logic to parse the ProxyConfiguration and return a new JenkinsProxyHelper should remain within JenkinsProxyHelper.fromJenkins(). While the data elements we use to create a new JenkinsProxyHelper come from the Jenkins ProxyConfiguration, the parsing we do for username and ntlmDomain is specific to our logic for using the proxy. If we do leave this logic within JenkinsProxyHelper, this wouldn't change the tests which use/test fromJenkins().
Thoughts?

Reply:
Yes, that's exactly what I'd suggest-- just a way to get the ProxyConfiguration, not to parse it-- the parsing logic makes sense to stay in the helper.